packages/api/models/rules.js

```diff
@@ -134,50 +134,61 @@ class Rule extends Manager {
     // Apply updates to updated rule item to be saved
     merge(updatedRuleItem, updates);
 
+    // Validate rule before kicking off workflows or adding event source mappings
+    await this.constructor.recordIsValid(updatedRuleItem, this.schema, this.removeAdditional);
+
     const stateChanged = (updates.state && updates.state !== original.state);
     const valueUpdated = (updates.rule
       && updates.rule.value !== original.rule.value);
 
+    updatedRuleItem = await this.updateRuleTrigger(updatedRuleItem, stateChanged, valueUpdated);
+
+    return super.update({ name: original.name }, updatedRuleItem,
+      fieldsToDelete);
+  }
+
+  async updateRuleTrigger(ruleItem, stateChanged, valueUpdated) {
+    let updatedRuleItem = cloneDeep(ruleItem);
+
     switch (updatedRuleItem.rule.type) {
-    case 'scheduled': {
-      const payload = await Rule.buildPayload(updatedRuleItem);
-      await this.addRule(updatedRuleItem, payload);
-      break;
-    }
-    case 'kinesis':
-      if (valueUpdated) {
-        await this.deleteKinesisEventSources(updatedRuleItem);
-        const updatedRuleItemArns = await this.addKinesisEventSources(updatedRuleItem);
-        updatedRuleItem = this.updateKinesisRuleArns(updatedRuleItem,
-          updatedRuleItemArns);
+      case 'scheduled': {
```

Comment:
When I check this out locally, ESlint is complaining about the spaces, though I don't know why